File: CoreJava/accountpolymorphism-app/src/com/techlabs/account/Account.java
package com.techlabs.account;

public abstract class Account {
	protected int accno;
	protected String name;
	protected double balance;

	public int getAccno() {
		return accno;
	}

	public String getName() {
		return name;
	}

	public double getBalance() {
		return balance;
	}

	protected Account(int accno, String name, double amount) {
		this.accno = accno;
		this.name = name;
		this.balance = amount;
	}
	
	public void deposit(double amount)
	{
		this.balance += amount;
	}
	public abstract void withdraw(double amount);
}

File: DotNET/C#/AccountPolymorphism/AccountPolymorphism/Account.cs
﻿using System;

namespace AccountPolymorphism
{
    abstract class Account
    {
        protected int _accountnum;
        protected int _balance;
        protected String _name;

        protected Account(int accountnum, int balance, String name)
        {
            _accountnum = accountnum;
            _balance = balance;
            _name = name;
        }

        public int AccountNum
        {
            get
            {
                return _accountnum;
            }
        }

        public int Balance
        {
            get
            {
                return _balance;
            }
            set
            {
                _balance = value;
            }
        }

        public String Name
        {
            get
            {
                return _name;
            }
        }

        public void deposit(int amount)
        {
            _balance += amount;
        }

        public abstract void withdraw(int amount);

    }
}
